### PR TITLE
Notify safekeeper readiness with systemd.

### PR DESCRIPTION
**Pull request by arssher**
To avoid downtime during deploy, as in busy regions initial load can currently take ~30s.

### CODE REVIEW

**Comment by arssher** [comment]
Tested it locally with unit file like
```
[Unit]
Description=Monitors room temperature by safekeeper

[Service]
# if you think it is a good idea, it isn't
# User=ars
ExecStart=/home/ars/neon/neon2/target/debug/safekeeper
WorkingDirectory=/home/ars/neon/neon2
Restart=on-failure
RestartSec=10
TimeoutStartSec=90s
# use sd_notify for determining readiness
Type=notify

[Install]
WantedBy=default.target
```
at `/home/ars/.config/systemd/user/safekeeper.service` with
`cargo build_testing && systemctl --user enable --now safekeeper && systemctl --user restart safekeeper.service`
, adding sleep before readiness notify.

**Review by petuhovskiy — APPROVED** [review]
Never used sd-notify before, let's deploy to staging and look

File: safekeeper/src/bin/safekeeper.rs [hunk 1]
```diff
@@ -434,6 +435,12 @@ async fn start_safekeeper(conf: SafeKeeperConf) -> Result<()> {
     let mut sigint_stream = signal(SignalKind::interrupt())?;
     let mut sigterm_stream = signal(SignalKind::terminate())?;
 
+    // Notify systemd that we are ready. This is important as currently loading
+    // timelines takes significant time (~30s in busy regions).
+    if let Err(e) = sd_notify::notify(true, &[NotifyState::Ready]) {
+        warn!("systemd notify failed: {:?}", e);
+    }
+
```
Comment (koivunej) [inline safekeeper/src/bin/safekeeper.rs:443]: How is the init ready at this point...? It is at least unclear to me where is the actual loading and startup, and why is this not at the end of that.

Comment (koivunej) [reply safekeeper/src/bin/safekeeper.rs:443]: Oki I had misunderstood what was needed; this is for rolling sequencial restarts, not a problem of incoherent responses while loading.

Loading is done in the `GlobalTimelines::init`.

**Pull request merged**